storage/range_command.go

```diff
@@ -39,59 +39,100 @@ import (
 // executeCmd switches over the method and multiplexes to execute the
 // appropriate storage API command. It returns an error and, for some calls
 // such as inconsistent reads, the intents they skipped.
-func (r *Range) executeCmd(batch engine.Engine, ms *engine.MVCCStats, args proto.Request, reply proto.Response) ([]proto.Intent, error) {
+func (r *Range) executeCmd(batch engine.Engine, ms *engine.MVCCStats, args proto.Request) (reply proto.Response, intents []proto.Intent, err error) {
+	// Ugh. Our callers expect non-nil replies even in the error case.
+	defer func() {
+		if reply == nil {
+			reply = args.CreateReply()
+		}
+	}()
+
 	// Verify key is contained within range here to catch any range split
 	// or merge activity.
 	header := args.Header()
 
-	if err := r.checkCmdHeader(header); err != nil {
-		reply.Header().SetGoError(err)
-		return nil, err
+	if err = r.checkCmdHeader(header); err != nil {
+		return
```

Comment:
I think we prefer `return nil, nil, err` instead of relying on named return values.